### ISSUE TITLE
update our watchers to support the new Watcher interface

### ISSUE THREAD

**Issue by hsluoyz** [casbin/jcasbin#206]
https://github.com/casbin/jcasbin/issues/201#issuecomment-864574337

**Comment by hsluoyz** [comment]
@shink

**Comment by shink** [comment]
@hsluoyz OK, I will.

**Comment by rongfengliang** [comment]
@shink  add  void setUpdateCallback(Consumer<String> func);  in Watcher interface as one default implement maybe  better 

 like this 

```code
 default void setUpdateCallback(Consumer<String> func) {};
```

so any  watcher  library deps  jcasbin nee't rewrite .just  upgrade   version
if one want  implement callback with args just override it

**Comment by shink** [comment]
@rongfengliang Yes, it's a good idea. But if the user calls this method and finds it blank, it may be misleading. We should remind the user to implement it. (There may exist third-party watchers.)
@hsluoyz What do you think?

**Comment by rongfengliang** [comment]
@shink  for default method  implement . we can show some log  message  contains  how to  implement  it

**Comment by shink** [comment]
@rongfengliang It's a good idea! Thank you for your suggestion.
